blocks/log.py

```diff
@@ -64,10 +69,3 @@ def previous_row(self):
     @property
     def last_epoch_row(self):
         return self[self.status['_epoch_ends'][-1]]
-
-    def to_dataframe(self):
-        """Convert a log into a :class:`.DataFrame`."""
-        if not PANDAS_AVAILABLE:
-            raise ImportError("The pandas library is not found. You can"
-                              " install it with pip.")
-        return DataFrame.from_dict(self, orient='index')
```

Comment:
Your first paragraph ignores the point that `to_dataframe` for other backends can take much more than 41 character. Then you say  "installations can be slow or fail", even though I did not argue in favour of automatically installing pandas. You are right that there will be an `ImportError`, but well, if a person calls `to_dataframe` he is expected to know what pandas is and how to get it. That is frustration will be close to zero. When I tried FFT convolution in Theano it was very natural for me to install a couple of required libraries. 

Right, there is a "lean and mean" motto. But it can be used to justify whatever one wants, up to the statement that Blocks is not necessary and everything should be done in pure Theano. If 80% users convert the log to a dataframe in order to analyze it, I doubt whether the place for this functionality should be in `blocks-extras`. 

I don't agree that 24 hours is okay for the PR that drops existing functionality. It would be okay for a new feature, but when you delete something it might cause serious frustration. Especially that you did not move this functionality to extras, you simply deleted it. 


Anyway, this discussion is turning lengthy, and I just want to make clear that here I care about two things:

- I would like to call for greater caution when something is removed from Blocks

- efficient routines for exporting the Blocks logs as dataframes should be provided, and I am not sure that `blocks-extras`  is the right place for it, because I find it very essential functionality. But at the very least they should be in `blocks-extras`, and the PR that removed them without moving there should be considered an example of excessive rush with merging.